### PR TITLE
help text

### PR DESCRIPTION
**Pull request by KevinRansom**
Update #help fsi command to display help text for package managers.

````
  F# Interactive directives:

    #r "file.dll";;                   // Reference (dynamically load) the given DLL
    #I "path";;                       // Add the given search path for referenced DLLs
    #load "file.fs" ...;;             // Load the given file(s) as if compiled and referenced
    #time ["on"|"off"];;              // Toggle timing on/off
    #help;;                           // Display help
    #r "nuget:FSharp.Data, 3.1.2";;   // Load Nuget Package 'FSharp.Data' version '3.1.2'
    #r "nuget:FSharp.Data";;          // Load Nuget Package 'FSharp.Data' with the highest version
    #quit;;                           // Exit

  F# Interactive command line options:
````
Update fsscript, to allow language configuration and quiet mode to improve testing.  Add tests.

### CODE REVIEW

**Comment by brettfo** [comment]
What about `#I "nuget:some-custom-feed-or-path"`?

**Comment by KevinRansom** [comment]
@brettfo , not implemented yet.

**Comment by KevinRansom** [comment]
/cc @brettfo, @cartermp 

please.

File: src/fsharp/FSharp.Compiler.Private.Scripting/FSharpScript.fs [hunk 1]
```diff
@@ -7,16 +7,38 @@ open System.Threading
...
+    let baseArgs = [|
+        typeof<FSharpScript>.Assembly.Location;
+        "--noninteractive";
+        "--targetprofile:" + computedProfile
+        if quiet then "--quiet"
+        match langVersion with
+        | LangVersion.V47 -> "--langversion:4.7"
+        | LangVersion.V50 -> "--langversion:5.0"
+        | LangVersion.Preview -> "--langversion:preview"
+        |]
```
Comment (cartermp) [inline src/fsharp/FSharp.Compiler.Private.Scripting/FSharpScript.fs:38]:
```suggestion
    let baseArgs =
        [|
            typeof<FSharpScript>.Assembly.Location;
            "--noninteractive";
            "--targetprofile:" + computedProfile
            if quiet then "--quiet"
            match langVersion with
            | LangVersion.V47 -> "--langversion:4.7"
            | LangVersion.V50 -> "--langversion:5.0"
            | LangVersion.Preview -> "--langversion:preview"
        |]
```

File: src/fsharp/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs [hunk 2]
```diff
@@ -163,6 +163,11 @@ type FSharpDependencyManager (outputDir:string option) =
 
     member _.Key = key
 
+    member _.HelpMessages = [|
+        sprintf """    #r "nuget:FSharp.Data, 3.1.2";;   // %s 'FSharp.Data' %s '3.1.2'""" (SR.loadNugetPackage()) (SR.version())
+        sprintf """    #r "nuget:FSharp.Data";;          // %s 'FSharp.Data' %s""" (SR.loadNugetPackage()) (SR.highestVersion())
+        |]
+
```
Comment (cartermp) [inline src/fsharp/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs:170]:
```suggestion
    member _.HelpMessages =
        [|
            sprintf """    #r "nuget:FSharp.Data, 3.1.2";;   // %s 'FSharp.Data' %s '3.1.2'""" (SR.loadNugetPackage()) (SR.version())
            sprintf """    #r "nuget:FSharp.Data";;          // %s 'FSharp.Data' %s""" (SR.loadNugetPackage()) (SR.highestVersion())
        |]

```

File: src/fsharp/Microsoft.DotNet.DependencyManager/DependencyProvider.fs [hunk 3]
```diff
@@ -301,6 +324,13 @@ type DependencyProvider (assemblyProbingPaths: AssemblyResolutionProbe, nativePr
     new (nativeProbingRoots: NativeResolutionProbe) =
         new DependencyProvider(Unchecked.defaultof<AssemblyResolutionProbe>, nativeProbingRoots)
 
+    /// Returns a formatted help messages for registered dependencymanagers for the host to present
+    member _.GetRegisteredDependencyManagerHelpText (compilerTools, outputDir, errorReport) = [|
+            let managers = RegisteredDependencyManagers compilerTools (Option.ofString outputDir) errorReport
+            for kvp in managers do
+                let dm = kvp.Value
+                yield! dm.HelpMessages
+        |]
```
Comment (cartermp) [inline src/fsharp/Microsoft.DotNet.DependencyManager/DependencyProvider.fs:333]:
```suggestion
    member _.GetRegisteredDependencyManagerHelpText (compilerTools, outputDir, errorReport) =
        [|
            let managers = RegisteredDependencyManagers compilerTools (Option.ofString outputDir) errorReport
            for kvp in managers do
                let dm = kvp.Value
                yield! dm.HelpMessages
        |]
```

**Pull request merged**